### ISSUE TITLE
Investigate "combine" as a new workflow type

### ISSUE THREAD

**Issue by ericvaandering** [dmwm/CRABServer#4942]
As discussed in the CRAB3 meeting yesterday and per my discussion some time ago with Giovanni Petrucciani, this is a good candidate for a workflow that CRAB3 might be able to handle. It's apparently common and benefits from running on a large scale.

**Comment by matz-e** [comment]
In short:

* The workflow I received runs many different `combine` commands in parallel via CRAB
* Input datacards are packed into the sandbox
* A list of commands is turned into a script selecting a command to run based on the jobnumber
* The specified `pset` is completely useless and does nothing
* Everything is run as `PrivateMC` to get the desired number of jobs

Questions/Suggestions:

* This seems to be best supported by a new workflow, as neither `PrivateMC` and `Analysis` seem a good fit?
* We could support `inputArguments` (list of list of arguments/list of strings?) to append to each invocation of `scriptExe`, replacing `inputDataset`?

Would there be enough demand to support this?  I have run workflows like this (not `combine`, though) at ND using condor/grid-control more than enough, but my workflows would needed local data access and used local python packages (in `~/.local/lib/…`).

**Comment by belforte** [comment]
Sorry for ignorance. What is 'combine' ? 
IIUC the description, the general use case is scriptExe for N jobs passing to it as arguments each time element n from a list of N. A list of N strings should do.  Maybe we can simply extend a bit semantic for script exe to capture this. Already privateMC is use for any workflow which has no input dataset,  which  is a bit ugly and we may want to fix, but it is a different story 
Sent from Stefano's mobile 
On Sep 2, 2015 15:40, Matthias Wolf <notifications@github.com> wrote:In short:

The workflow I received runs many different combine commands in parallel via CRABInput datacards are packed into the sandboxA list of commands is turned into a script selecting a command to run based on the jobnumberThe specified pset is completely useless and does nothingEverything is run as PrivateMC to get the desired number of jobs

Questions/Suggestions:

This seems to be best supported by a new workflow, as neither PrivateMC and Analysis seem a good fit?We could support inputArguments (list of list of arguments/list of strings?) to append to each invocation of scriptExe, replacing inputDataset?

Would there be enough demand to support this?  I have run workflows like this (not combine, though) at ND using condor/grid-control more than enough, but my workflows would needed local data access and used local python packages (in ~/.local/lib/…).

**Comment by matz-e** [comment]
`combine` is the CMS limit setting and statistical analysis tool.

**Comment by belforte** [comment]
Can we conclude that this is mostly a documentation and provide-example issue ?
About the workflow names, we need to start from a definition of the current ones, since PrivateMC may imply more than "no input dataset" but I am not sure. If that were all... I'd in favour of renaming those plugins.

**Comment by belforte** [comment]
see also discussion here. We should consider these two issues as combined.
https://github.com/dmwm/CRABServer/issues/4857

**Comment by matz-e** [comment]
In my opinion it's more than documentation, because you essentially hack CRAB and abuse the `PrivateMC`.  Putting the burden on the user to implement multi-crab, running custom commands in a reasonable fashion and then saying it is supported seems pretty wrong to me.

**Comment by ericvaandering** [comment]
:+1:

**Comment by belforte** [comment]
I suspect we do not understand each other. Maybe you want to write down how you would address this ?
Seems to me we agree on  `inputArguments` (list of list of arguments/list of strings?) and that we want a workflow (plugin) type other than `PrivateMC` for a generic "no input dataset to look up" use case.

**Comment by mmascher** [comment]
Yes, we agree on this. I think your sentence "Can we conclude that this is
mostly a documentation and provide-example issue ?" is confusing, this
requires a little bit of development

**Comment by mmascher** [comment]
This depends on making the PSet optional (for scriptexe)

**Comment by belforte** [comment]
that's (making the PSet optional ) is #4922
